Sources/EventProducer/TidalEventSender.swift

```diff
@@ -42,19 +43,22 @@ public final class TidalEventSender: EventSender {
 		self.eventSubmitter = EventSubmitter(eventsQueue: eventQueue, monitoring: monitoring)
 		
 		self.fileManager = FileManagerHelper.shared
-		
-		self.scheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
-		self.monitoringScheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
 	}
 
 	public func config(_ config: EventConfig) {
 		self.config = config
+		
+		// In case of reentrancy we invalidate schedulers before creating new ones
+		self.scheduler?.invalidateTimer()
+		self.monitoringScheduler?.invalidateTimer()
+
 		self.scheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
 		self.monitoringScheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
 
 		self.start()
 	}
 	
+	@available(*, deprecated)
 	public func updateConfiguration(_ config: EventConfig) {
```

Comment:
I still vote to kill it, as it's not bringing value in the current state (on the contrary, it breaks the logging), and its removal will be a breaking change at some point, so better to remote it sooner rather than later.

Reply:
I also thought to remove it tbh, but I am not sure what to do with these tests:

https://github.com/tidal-music/tidal-sdk-ios/blob/9a6383212dda76570e3fa24ddb9435af3b758ff1/Tests/EventProducerTests/EventsTests/EventsTests.swift#L79-L108

https://github.com/tidal-music/tidal-sdk-ios/blob/9a6383212dda76570e3fa24ddb9435af3b758ff1/Tests/EventProducerTests/EventsTests/EventsTests.swift#L130-L161

https://github.com/tidal-music/tidal-sdk-ios/blob/9a6383212dda76570e3fa24ddb9435af3b758ff1/Tests/EventProducerTests/EventsTests/EventsTests.swift#L163-L191

They all call `updateConfiguration` function. I could for example, call `config` function again instead...

Or maybe you have another idea?

Reply:
@ksloginov still waiting for your input... 🙂

Reply:
Changed [here](https://github.com/tidal-music/tidal-sdk-ios/pull/36/commits/0817ec3da354cc04705cfb1c3f7ae7f9c8eb7304)